sdk/src/manifest_assertion.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{
+    assertion::{AssertionBase, AssertionDecodeError},
+    error::{Error, Result},
+};
+
+use serde::{de::DeserializeOwned, Deserialize, Serialize}; //,  Deserializer, Serializer};
+use serde_json::Value;
+
+/// Assertions in C2PA can be stored in several formats
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub enum ManifestAssertionKind {
+    Cbor,
+    Json,
+    Binary,
+    Uri,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(untagged)]
+enum ManifestData {
+    Json(Value),     // { label: String, instance: usize, data: Value },
+    Binary(Vec<u8>), // ) { label: String, instance: usize, data: Value },
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+/// A labeled container for an Assertion value in a Manifest
+pub struct ManifestAssertion {
+    /// An assertion label in reverse domain format
+    label: String,
+    /// The data of the assertion as Value
+    //#[serde(deserialize_with = "manifest_data_deserialize", serialize_with = "manifest_data_serialize")]
+    data: ManifestData,
+    /// There can be more than one assertion for any label
+    #[serde(skip_serializing_if = "Option::is_none")]
+    instance: Option<usize>,
```

Comment:
I'm guessing this will come back in the ManifestStore payload automatically?

Reply:
yes, the manifest store also uses manifest_assertion and thus ManifestData.